src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_LoongArch64/LoongArch64ReadyToRunHelperNode.cs

```diff
@@ -48,30 +48,37 @@ protected override void EmitCode(NodeFactory factory, ref LoongArch64Emitter enc
                 case ReadyToRunHelperId.GetThreadStaticBase:
                     {
                         MetadataType target = (MetadataType)Target;
-                        encoder.EmitMOV(encoder.TargetRegister.Arg2, factory.TypeThreadStaticIndex(target));
-
-                        // First arg: address of the TypeManager slot that provides the helper with
-                        // information about module index and the type manager instance (which is used
-                        // for initialization on first access).
-                        encoder.EmitLD(encoder.TargetRegister.Arg0, encoder.TargetRegister.Arg2, 0);
-
-                        // Second arg: index of the type in the ThreadStatic section of the modules
-                        encoder.EmitLD(encoder.TargetRegister.Arg1, encoder.TargetRegister.Arg2, factory.Target.PointerSize);
-
-                        if (!factory.PreinitializationManager.HasLazyStaticConstructor(target))
+                        if (factory.TypeThreadStaticIndex(target) is TypeThreadStaticIndexNode ti && ti.IsInlined)
                         {
-                            encoder.EmitJMP(factory.HelperEntrypoint(HelperEntrypoint.GetThreadStaticBaseForType));
+                            throw new NotImplementedException();
```

Comment:
Does this need TODO?

Reply:
The inlining optimization was originally done for ARM64/X64 in ILC (for Release build settings). It was later mostly moved to JIT. I am not sure if it can still practically happen.